spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkFileWriterFactory.java

```diff
@@ -68,119 +54,34 @@ class SparkFileWriterFactory extends BaseFileWriterFactory<InternalRow> {
     super(
         table,
         dataFileFormat,
+        SparkObjectModels.SPARK_OBJECT_MODEL,
         dataSchema,
         dataSortOrder,
         deleteFileFormat,
         equalityFieldIds,
         equalityDeleteRowSchema,
         equalityDeleteSortOrder,
-        positionDeleteRowSchema);
-
-    this.dataSparkType = dataSparkType;
-    this.equalityDeleteSparkType = equalityDeleteSparkType;
-    this.positionDeleteSparkType = positionDeleteSparkType;
-    this.writeProperties = writeProperties != null ? writeProperties : ImmutableMap.of();
+        positionDeleteRowSchema,
+        writeProperties,
+        dataSparkType == null
+            ? dataSchema == null ? null : SparkSchemaUtil.convert(dataSchema)
+            : dataSparkType,
+        equalityDeleteSparkType == null
+            ? equalityDeleteRowSchema == null
+                ? null
+                : SparkSchemaUtil.convert(equalityDeleteRowSchema)
+            : equalityDeleteSparkType,
+        positionDeleteSparkType == null
+            ? positionDeleteRowSchema == null
+                ? null
+                : SparkSchemaUtil.convert(DeleteSchemaUtil.posDeleteSchema(positionDeleteRowSchema))
+            : positionDeleteSparkType);
```

Comment:
This is very hard to read and maintain with so many nested ternary operators.

Reply:
Refactored to a common method